File: src/main/java/com/baulsupp/oksocial/services/squareup/SquareUpUtil.java
package com.baulsupp.oksocial.services.squareup;

import com.google.common.collect.Sets;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.Set;
import okhttp3.Request;

public class SquareUpUtil {
  private SquareUpUtil() {
  }

  public static final Set<String> API_HOSTS =
      Collections.unmodifiableSet(Sets.newHashSet(
          "connect.squareup.com")
      );

  public static Collection<String> ALL_PERMISSIONS = Arrays.asList(
      "MERCHANT_PROFILE_READ",
      "PAYMENTS_READ",
      "SETTLEMENTS_READ",
      "BANK_ACCOUNTS_READ"
  );

  public static Request apiRequest(String s, Request.Builder requestBuilder) {
    return requestBuilder.url("https://connect.squareup.com" + s).build();
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/squareup/SquareUpUtil.kt
package com.baulsupp.oksocial.services.squareup

import okhttp3.Request
import java.util.Arrays

object SquareUpUtil {

  val API_HOSTS = setOf((
      "connect.squareup.com")
  )

  var ALL_PERMISSIONS: Collection<String> = Arrays.asList(
      "MERCHANT_PROFILE_READ",
      "PAYMENTS_READ",
      "SETTLEMENTS_READ",
      "BANK_ACCOUNTS_READ"
  )

  fun apiRequest(s: String, requestBuilder: Request.Builder): Request {
    return requestBuilder.url("https://connect.squareup.com" + s).build()
  }
}
